core/pkg/resourcehandler/localgitrepository.go

```diff
@@ -0,0 +1,107 @@
+package resourcehandler
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+type LocalGitRepository struct {
+	repo   *git.Repository
+	head   *plumbing.Reference
+	config *config.Config
+}
+
+type GitCommit struct {
+	hash        string
+	authorName  string
+	authorEmail string
+	message     string
+	date        string
```

Comment:
Is it possible to use `time.Time`? What is the time format you can extract from the local git?

Reply:
Yes, `commit.Author.When` is of type `time.Time`. I will change it